<html><head>
		<title>似然树构建</title>
		<base href="..\../">
		<meta id="root-path" root-path="..\../">
		<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes, minimum-scale=1.0, maximum-scale=5.0">
		<meta charset="UTF-8">
		<meta name="description" content="长脑子了 - 似然树构建">
		<meta property="og:title" content="似然树构建">
		<meta property="og:description" content="长脑子了 - 似然树构建">
		<meta property="og:type" content="website">
		<meta property="og:url" content="软件操作/分析软件/似然树构建.html">
		<meta property="og:image" content="https://mmbiz.qpic.cn/mmbiz_png/bjvWWMcEDllRjKyOZo0rc7kWeAKLnYaKuA5Kqv6kCvH1S42cNvsMwg1A6XwEQkH0wkvQfHvPA96Y8ZaU2hZulQ/640?wx_fmt=png&amp;tp=wxpic&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1">
		<meta property="og:site_name" content="长脑子了">
		<link rel="alternate" type="application/rss+xml" title="RSS Feed" href="lib/rss.xml"><script async="" id="webpage-script" src="lib/scripts/webpage.js" onload="this.onload=null;this.setAttribute(&quot;loaded&quot;, &quot;true&quot;)"></script><script type="module" async="" id="graph-view-script" src="lib/scripts/graph-view.js"></script><script async="" id="graph-wasm-script" src="lib/scripts/graph-wasm.js" onload="this.onload=null;this.setAttribute(&quot;loaded&quot;, &quot;true&quot;)"></script><script async="" id="graph-render-worker-script" src="lib/scripts/graph-render-worker.js" onload="this.onload=null;this.setAttribute(&quot;loaded&quot;, &quot;true&quot;)"></script><script async="" id="tinycolor-script" src="lib/scripts/tinycolor.js" onload="this.onload=null;this.setAttribute(&quot;loaded&quot;, &quot;true&quot;)"></script><script async="" id="pixi-script" src="https://cdnjs.cloudflare.com/ajax/libs/pixi.js/7.4.0/pixi.min.js" onload="this.onload=null;this.setAttribute(&quot;loaded&quot;, &quot;true&quot;)"></script><script async="" id="minisearch-script" src="https://cdn.jsdelivr.net/npm/minisearch@6.3.0/dist/umd/index.min.js" onload="this.onload=null;this.setAttribute(&quot;loaded&quot;, &quot;true&quot;)"></script><link rel="icon" href="lib/media/favicon.png"><script async="" id="graph-data-script" src="lib/scripts/graph-data.js" onload="this.onload=null;this.setAttribute(&quot;loaded&quot;, &quot;true&quot;)"></script><style>body{--line-width:40em;--line-width-adaptive:40em;--file-line-width:40em;--sidebar-width:min(20em, 80vw);--collapse-arrow-size:11px;--tree-horizontal-spacing:0.6em;--tree-vertical-spacing:0.6em;--sidebar-margin:12px}.sidebar{height:100%;min-width:calc(var(--sidebar-width) + var(--divider-width-hover));max-width:calc(var(--sidebar-width) + var(--divider-width-hover));font-size:14px;z-index:10;position:relative;overflow:hidden;transition:min-width ease-in-out,max-width ease-in-out;transition-duration:.2s;contain:size}.sidebar-left{left:0}.sidebar-right{right:0}.sidebar.is-collapsed{min-width:0;max-width:0}body.floating-sidebars .sidebar{position:absolute}.sidebar-content{height:100%;min-width:calc(var(--sidebar-width) - var(--divider-width-hover));top:0;padding:var(--sidebar-margin);padding-top:4em;line-height:var(--line-height-tight);background-color:var(--background-secondary);transition:background-color,border-right,border-left,box-shadow;transition-duration:var(--color-fade-speed);transition-timing-function:ease-in-out;position:absolute;display:flex;flex-direction:column}.sidebar:not(.is-collapsed) .sidebar-content{min-width:calc(max(100%,var(--sidebar-width)) - 3px);max-width:calc(max(100%,var(--sidebar-width)) - 3px)}.sidebar-left .sidebar-content{left:0;border-top-right-radius:var(--radius-l);border-bottom-right-radius:var(--radius-l)}.sidebar-right .sidebar-content{right:0;border-top-left-radius:var(--radius-l);border-bottom-left-radius:var(--radius-l)}.sidebar:has(.sidebar-content:empty):has(.topbar-content:empty){display:none}.sidebar-topbar{height:2em;width:var(--sidebar-width);top:var(--sidebar-margin);padding-inline:var(--sidebar-margin);z-index:1;position:fixed;display:flex;align-items:center;transition:width ease-in-out;transition-duration:inherit}.sidebar.is-collapsed .sidebar-topbar{width:calc(2.3em + var(--sidebar-margin) * 2)}.sidebar .sidebar-topbar.is-collapsed{width:0}.sidebar-left .sidebar-topbar{left:0}.sidebar-right .sidebar-topbar{right:0}.topbar-content{overflow:hidden;overflow:clip;width:100%;height:100%;display:flex;align-items:center;transition:inherit}.sidebar.is-collapsed .topbar-content{width:0;transition:inherit}.clickable-icon.sidebar-collapse-icon{background-color:transparent;color:var(--icon-color-focused);padding:0!important;margin:0!important;height:100%!important;width:2.3em!important;margin-inline:0.14em!important;position:absolute}.sidebar-left .clickable-icon.sidebar-collapse-icon{transform:rotateY(180deg);right:var(--sidebar-margin)}.sidebar-right .clickable-icon.sidebar-collapse-icon{transform:rotateY(180deg);left:var(--sidebar-margin)}.clickable-icon.sidebar-collapse-icon svg.svg-icon{width:100%;height:100%}.sidebar-section-header{margin:0 0 1em 0;text-transform:uppercase;letter-spacing:.06em;font-weight:600}body{transition:background-color var(--color-fade-speed) ease-in-out}.webpage-container{display:flex;flex-direction:row;height:100%;width:100%;align-items:stretch;justify-content:center}.document-container{opacity:1;flex-basis:100%;max-width:100%;width:100%;height:100%;display:flex;flex-direction:column;align-items:center;transition:opacity .2s ease-in-out;contain:inline-size}.hide{opacity:0;transition:opacity .2s ease-in-out}.document-container>.markdown-preview-view{margin:var(--sidebar-margin);margin-bottom:0;width:100%;width:-webkit-fill-available;width:-moz-available;width:fill-available;background-color:var(--background-primary);transition:background-color var(--color-fade-speed) ease-in-out;border-top-right-radius:var(--window-radius,var(--radius-m));border-top-left-radius:var(--window-radius,var(--radius-m));overflow-x:hidden!important;overflow-y:auto!important;display:flex!important;flex-direction:column!important;align-items:center!important;contain:inline-size}.document-container>.markdown-preview-view>.markdown-preview-sizer{padding-bottom:80vh!important;width:100%!important;max-width:var(--line-width)!important;flex-basis:var(--line-width)!important;transition:background-color var(--color-fade-speed) ease-in-out;contain:inline-size}.markdown-rendered img:not([width]),.view-content img:not([width]){max-width:100%;outline:0}.document-container>.view-content.embed{display:flex;padding:1em;height:100%;width:100%;align-items:center;justify-content:center}.document-container>.view-content.embed>*{max-width:100%;max-height:100%;object-fit:contain}:has(> :is(.math,table)){overflow-x:auto!important}.document-container>.view-content{overflow-x:auto;contain:content;padding:0;margin:0;height:100%}.scroll-highlight{position:absolute;width:100%;height:100%;pointer-events:none;z-index:1000;background-color:hsla(var(--color-accent-hsl),.25);opacity:0;padding:1em;inset:50%;translate:-50% -50%;border-radius:var(--radius-s)}</style><script defer="">async function loadIncludes(){if("file:"!=location.protocol){let e=document.querySelectorAll("include");for(let t=0;t<e.length;t++){let o=e[t],l=o.getAttribute("src");try{const e=await fetch(l);if(!e.ok){console.log("Could not include file: "+l),o?.remove();continue}let t=await e.text(),n=document.createRange().createContextualFragment(t),i=Array.from(n.children);for(let e of i)e.classList.add("hide"),e.style.transition="opacity 0.5s ease-in-out",setTimeout((()=>{e.classList.remove("hide")}),10);o.before(n),o.remove(),console.log("Included file: "+l)}catch(e){o?.remove(),console.log("Could not include file: "+l,e);continue}}}else{if(document.querySelectorAll("include").length>0){var e=document.createElement("div");e.id="error",e.textContent="Web server exports must be hosted on an http / web server to be viewed correctly.",e.style.position="fixed",e.style.top="50%",e.style.left="50%",e.style.transform="translate(-50%, -50%)",e.style.fontSize="1.5em",e.style.fontWeight="bold",e.style.textAlign="center",document.body.appendChild(e),document.querySelector(".document-container")?.classList.remove("hide")}}}document.addEventListener("DOMContentLoaded",(()=>{loadIncludes()}));let isFileProtocol="file:"==location.protocol;function waitLoadScripts(e,t){let o=e.map((e=>document.getElementById(e+"-script"))),l=0;!function e(){let n=o[l];l++,n&&"true"!=n.getAttribute("loaded")||l<o.length&&e(),l<o.length?n.addEventListener("load",e):t()}()}</script><link rel="stylesheet" href="lib/styles/obsidian.css"><link rel="stylesheet" href="lib/styles/theme.css"><link rel="preload" href="lib/styles/global-variable-styles.css" as="style" onload="this.onload=null;this.rel='stylesheet'"><noscript><link rel="stylesheet" href="lib/styles/global-variable-styles.css"></noscript><link rel="preload" href="lib/styles/main-styles.css" as="style" onload="this.onload=null;this.rel='stylesheet'"><noscript><link rel="stylesheet" href="lib/styles/main-styles.css"></noscript></head><body class="publish css-settings-manager theme-dark show-inline-title show-ribbon"><script defer="">let theme=localStorage.getItem("theme")||(window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light");"dark"==theme?(document.body.classList.add("theme-dark"),document.body.classList.remove("theme-light")):(document.body.classList.add("theme-light"),document.body.classList.remove("theme-dark")),window.innerWidth<480?document.body.classList.add("is-phone"):window.innerWidth<768?document.body.classList.add("is-tablet"):window.innerWidth<1024?document.body.classList.add("is-small-screen"):document.body.classList.add("is-large-screen")</script><div class="webpage-container workspace"><div class="sidebar-left sidebar"><div class="sidebar-handle"></div><div class="sidebar-topbar"><div class="topbar-content"><label class="theme-toggle-container" for="theme_toggle"><input class="theme-toggle-input" type="checkbox" id="theme_toggle"><div class="toggle-background"></div></label></div><div class="clickable-icon sidebar-collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="100%" height="100%" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="3" stroke-linecap="round" stroke-linejoin="round" class="svg-icon"><path d="M21 3H3C1.89543 3 1 3.89543 1 5V19C1 20.1046 1.89543 21 3 21H21C22.1046 21 23 20.1046 23 19V5C23 3.89543 22.1046 3 21 3Z"></path><path d="M10 4V20"></path><path d="M4 7H7"></path><path d="M4 10H7"></path><path d="M4 13H7"></path></svg></div></div><div class="sidebar-content"><div class="search-input-container"><input enterkeyhint="search" type="search" spellcheck="false" placeholder="Search..."><div class="search-input-clear-button" aria-label="Clear search"></div></div><include src="lib/html/file-tree.html"></include></div><script defer="">let ls = document.querySelector(".sidebar-left"); ls.classList.add("is-collapsed"); if (window.innerWidth > 768) ls.classList.remove("is-collapsed"); ls.style.setProperty("--sidebar-width", localStorage.getItem("sidebar-left-width"));</script></div><div class="document-container markdown-reading-view hide"><div class="markdown-preview-view markdown-rendered allow-fold-headings allow-fold-lists is-readable-line-width"><style id="MJX-CHTML-styles"></style><div class="markdown-preview-sizer markdown-preview-section"><h1 class="page-title heading inline-title" id="似然树构建"><p dir="auto">似然树构建</p></h1><div class="heading-wrapper"><h3 data-heading="构建进化树的软件" dir="auto" class="heading" id="构建进化树的软件"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>构建进化树的软件</h3><div class="heading-children"><div><p dir="auto">构建进化树的方法和软件很多，前面我们讲解构<a data-tooltip-position="top" aria-label="http://mp.weixin.qq.com/s?__biz=MzIyNzIyNTczNA==&amp;mid=2247491242&amp;idx=3&amp;sn=b4a8232832d08bf1c08c9291503354a8&amp;chksm=e8652743df12ae55c8b120eee15e9e6774ac7714ad5467b1e1046ddf7e7740729d6e3f0d889e&amp;scene=21#wechat_redirect" rel="noopener" class="external-link" href="http://mp.weixin.qq.com/s?__biz=MzIyNzIyNTczNA==&amp;mid=2247491242&amp;idx=3&amp;sn=b4a8232832d08bf1c08c9291503354a8&amp;chksm=e8652743df12ae55c8b120eee15e9e6774ac7714ad5467b1e1046ddf7e7740729d6e3f0d889e&amp;scene=21#wechat_redirect" target="_blank">建进化树的原理</a>时提过，最准确的方法为贝叶斯方法，但是贝叶斯的方法计算量太大太耗时，对于大量的数据不适用，其次就是最大似然法，这里我们解释三种利用最大似然法构建进化树的软件：fasttree，iqtree2，RAxML-ng.输入的数据为做群体遗传进化得到的SNP数据。  </p></div><div><p dir="auto"><img alt="图片" src="https://mmbiz.qpic.cn/mmbiz_png/bjvWWMcEDllRjKyOZo0rc7kWeAKLnYaKuA5Kqv6kCvH1S42cNvsMwg1A6XwEQkH0wkvQfHvPA96Y8ZaU2hZulQ/640?wx_fmt=png&amp;tp=wxpic&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" referrerpolicy="no-referrer"></p></div><div><p dir="auto"><strong>数据准备</strong></p></div><div><p dir="auto">做完重测序分析后，我们可以得到包含 SNP的变异结果vcf文件作为输入文件。详见课程：<a data-tooltip-position="top" aria-label="http://mp.weixin.qq.com/s?__biz=MzIyNzIyNTczNA==&amp;mid=2247490638&amp;idx=1&amp;sn=bc420d5bea451c9998162b2616276791&amp;chksm=e86525a7df12acb1527d54312c9d08c6625dd0bf12b3e8fc64c2a187442af55c73f59b21cfbf&amp;scene=21#wechat_redirect" rel="noopener" class="external-link" href="http://mp.weixin.qq.com/s?__biz=MzIyNzIyNTczNA==&amp;mid=2247490638&amp;idx=1&amp;sn=bc420d5bea451c9998162b2616276791&amp;chksm=e86525a7df12acb1527d54312c9d08c6625dd0bf12b3e8fc64c2a187442af55c73f59b21cfbf&amp;scene=21#wechat_redirect" target="_blank">重测序数据分析课程</a>，然后利用omicsclass/pop-evol-gwas:v1.3 镜像进行前期数据准备与后续进化树构建分析：</p></div><div><pre><code>#运行环境准备：docker镜像启动#镜像下载：docker&nbsp;pull&nbsp;omicsclass/pop-evol-gwas:v1.3#启动遗传进化镜像docker&nbsp;run&nbsp;--rm&nbsp;-it&nbsp;-m&nbsp;4G&nbsp;--cpus&nbsp;1&nbsp;&nbsp;-v&nbsp;D:\pop:/work&nbsp;omicsclass/pop-evol-gwas:v1.3#对vcf文件进行数据过滤vcftools&nbsp;--gzvcf&nbsp;all.varFilter.vcf.gz&nbsp;--recode&nbsp;--recode-INFO-all&nbsp;--stdout&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--maf&nbsp;0.05&nbsp;&nbsp;--max-missing&nbsp;0.7&nbsp;&nbsp;--minDP&nbsp;4&nbsp;&nbsp;--maxDP&nbsp;1000&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;\&nbsp;&nbsp;&nbsp;&nbsp;--minQ&nbsp;30&nbsp;--minGQ&nbsp;0&nbsp;--min-alleles&nbsp;2&nbsp;&nbsp;--max-alleles&nbsp;2&nbsp;--remove-indels&nbsp;|gzip&nbsp;-&nbsp;&gt;&nbsp;clean.vcf.gz#利用tassel软件对文件进行排序run_pipeline.pl&nbsp;-Xmx30G&nbsp;&nbsp;-SortGenotypeFilePlugin&nbsp;-inputFile&nbsp;clean.vcf.gz&nbsp;\&nbsp;&nbsp;&nbsp;&nbsp;-outputFile&nbsp;clean.sorted.vcf.gz&nbsp;-fileType&nbsp;VCF#vcf文件格式转换成Phylip格式，用于后续构建进化树run_pipeline.pl&nbsp;&nbsp;-Xmx5G&nbsp;-importGuess&nbsp;&nbsp;$workdir/00.filter/clean.sorted.vcf.gz&nbsp;&nbsp;\&nbsp;&nbsp;&nbsp;&nbsp;-ExportPlugin&nbsp;-saveAs&nbsp;supergene.phy&nbsp;-format&nbsp;Phylip_Inter
</code><button class="copy-code-button">复制</button></pre></div></div></div><div class="heading-wrapper"><h3 data-heading="1. FastTree 构建进化树" dir="auto" class="heading" id="1._FastTree_构建进化树"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>1. FastTree 构建进化树</h3><div class="heading-children"><div><p dir="auto">FastTree 是基于最大似然法构建进化树的软件，它最大的特点就是运行速度快，支持几百万条序列的建树任务。但是fasttree不支持bootstrap检验以及支持的替换模型有限。官网如下：<a rel="noopener" class="external-link" href="http://www.microbesonline.org/fasttree/" target="_blank">http://www.microbesonline.org/fasttree/</a></p></div><div class="heading-wrapper"><h5 data-heading="替换模型选择" dir="auto" class="heading" id="替换模型选择"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>替换模型选择</h5><div class="heading-children"><div><p dir="auto">FastTree 支持核酸和蛋白的进化树构建，对于核酸，可选的替换模型包括以下几种：JC（Jukes-Cantor）、GTR（generalized time-reversible），默认的模型为JC。对于蛋白质，可选的替换模型包括以下几种：JTT (Jones-Taylor-Thornton 1992)、LG(Le and Gascuel 2008)、WAG(Whelan &amp; Goldman 2001) 默认的模型为JTT。FastTree要求输入的多序列比对结果为FASTA或者Phylip格式。<br>
构建进化树命令举例：</p></div><div><pre><code>fasttree&nbsp;-nt&nbsp;-gtr&nbsp;&nbsp;supergene.fa&nbsp;&nbsp;&nbsp;&gt;&nbsp;&nbsp;fasttree.nwk
</code><button class="copy-code-button">复制</button></pre></div><div><p dir="auto">FastTree 其他命令举例：</p></div><div><pre><code>#对于蛋白质的进化树构建，基本用法如下fasttree&nbsp;protein.fasta&nbsp;&gt;&nbsp;tree#也可以选择LG或者WAG替换模型，用法如下fasttree&nbsp;-lg&nbsp;protein.fasta&nbsp;&gt;&nbsp;treefasttree&nbsp;-wag&nbsp;protein.fasta&nbsp;&gt;&nbsp;tree#对于核酸序列，基本用法如下fasttree&nbsp;-nt&nbsp;nucleotide.fasta&nbsp;&gt;&nbsp;tree#也可以选择GTR替换模型，用法如下fasttree&nbsp;-nt&nbsp;&nbsp;-gtr&nbsp;nucleotide.fasta&nbsp;&gt;&nbsp;tree
</code><button class="copy-code-button">复制</button></pre></div></div></div></div></div><div class="heading-wrapper"><h3 data-heading="2.&nbsp;IQ-tree 构建进化树" dir="auto" class="heading" id="2.&nbsp;IQ-tree_构建进化树"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>2.&nbsp;IQ-tree 构建进化树</h3><div class="heading-children"><div><p dir="auto">IQ-tree也是一款最大似然法构建进化树的软件，目前IQ-tree已经更新到2.0版本功能和性能也有很大的提升，主要有四大功能，高效建树（efficient tree reconstruction），模型选择（modelfinder: fast and accurate model selection），超快bootstrap（ultrafast bootstrap approximation），大型数据（big data analysis），以上特点特别适用于高通量测序的大量SNP构建进化树。</p></div><div class="heading-wrapper"><h5 data-heading="模型选择" dir="auto" class="heading" id="模型选择"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>模型选择</h5><div class="heading-children"><div><p dir="auto">构建进化树有很多模型对于初学者往往不知道那种模型最适合，iqtree提供自动模型选择功能，使用的软件为modelfinder。Modelfinder是一款速度超快的自动最佳模型选择软件。其在保证准确性的情况下，速度上比jmodeltest（针对DNA）和prottest（针对蛋白）要快100倍（ModelFinder is up to 100 times faster than jModelTest/ProtTest.），使用命令举例：</p></div><div><pre><code>#自动选择最佳模型并构建进化树：-m&nbsp;MFPiqtree&nbsp;-s&nbsp;supergene.phy&nbsp;-m&nbsp;MFP#只是想找最佳模型不构建进化树：iqtree&nbsp;-s&nbsp;example.phy&nbsp;-m&nbsp;MF#查找模型计算过程：ModelFinder&nbsp;will&nbsp;test&nbsp;up&nbsp;to&nbsp;546&nbsp;protein&nbsp;models&nbsp;(sample&nbsp;size:&nbsp;36415)&nbsp;...&nbsp;No.&nbsp;Model&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-LnL&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;df&nbsp;&nbsp;AIC&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;AICc&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;BIC&nbsp;&nbsp;1&nbsp;&nbsp;LG&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;10134094.366&nbsp;349&nbsp;20268886.731&nbsp;20268893.505&nbsp;20271854.186&nbsp;&nbsp;2&nbsp;&nbsp;LG+I&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;10133927.677&nbsp;350&nbsp;20268555.354&nbsp;20268562.167&nbsp;20271531.312&nbsp;&nbsp;3&nbsp;&nbsp;LG+G4&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;10043239.052&nbsp;350&nbsp;20087178.104&nbsp;20087184.917&nbsp;20090154.062&nbsp;&nbsp;4&nbsp;&nbsp;LG+I+G4&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;10043175.024&nbsp;351&nbsp;20087052.048&nbsp;20087058.900&nbsp;20090036.508&nbsp;&nbsp;5&nbsp;&nbsp;LG+R2&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;10063911.721&nbsp;351&nbsp;20128525.442&nbsp;20128532.294&nbsp;20131509.902&nbsp;&nbsp;6&nbsp;&nbsp;LG+R3&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;10045448.117&nbsp;353&nbsp;20091602.235&nbsp;20091609.165&nbsp;20094603.701
</code><button class="copy-code-button">复制</button></pre></div><div><p dir="auto">MFP为ModelFinder Plus的缩写，该参数使程序执行ModelFinder选择最适模型并完成建树分析。ModelFinder为许多不同的模型计算初始简约树的逻辑概率，并产生Akaike information criterion (AIC),<em> corrected Akaike information criterion</em> (AICc), and<em> the Bayesian information criterion</em> (BIC)三个结果标准值，通常ModelFinder选择BIC分数最低的模型（当然也可以指定AIC和AICc通过指定选项-AIC或者-AICc)。如果你想节约时间，可指定选择的模型及编码参数，例如:从WAG ,LG ,JTT 核苷酸替代模型里选一个: -mset WAG,LG,JTT；在+G和+I，以及+I+G三个里面选择rate ：-mrate G,I,I+G；heterogeneity参数：-mfreq FU,F</p></div><div><p dir="auto">命令行如下：</p></div><div><pre><code>iqtree&nbsp;-s&nbsp;example.phy&nbsp;-m&nbsp;MPF&nbsp;-mset&nbsp;WAG,LG,JTT&nbsp;&nbsp;-mrate&nbsp;G,I,I+G&nbsp;-mfreq&nbsp;FU,F
</code><button class="copy-code-button">复制</button></pre></div><div class="heading-wrapper"><h6 data-heading="指定模型参数设置格式：-m MODEL+FreqType+RateType，" dir="auto" class="heading" id="指定模型参数设置格式：-m_MODEL+FreqType+RateType，"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>指定模型参数设置格式：-m MODEL+FreqType+RateType，</h6><div class="heading-children"><div><pre><code>MODEL：model name+FreqType：(可选项)frequency type+RateType：(可选项)rate heterogeneity type
</code><button class="copy-code-button">复制</button></pre></div></div></div></div></div><div class="heading-wrapper"><h5 data-heading="替换模型MODEL包括：" dir="auto" class="heading" id="替换模型MODEL包括："><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>替换模型MODEL包括：</h5><div class="heading-children"><div><ul>
<li data-line="0" dir="auto">DNA模型:<br>
</li>
</ul></div><div><p dir="auto">JC/JC69, F81, K2P/K80, HKY/HKY85, TN/TrN/TN93, TNe,<br>
K3P/K81, K81u, TPM2, TPM2u, TPM3, TPM3u, TIM, TIMe,<br>
TIM2, TIM2e, TIM3, TIM3e, TVM, TVMe, SYM, GTR and 6-digit</p></div><div><ul>
<li data-line="0" dir="auto">蛋白质模型：<br>
</li>
</ul></div><div><p dir="auto">BLOSUM62, cpREV, Dayhoff, DCMut, FLU, HIVb, HIVw, JTT,<br>
JTTDCMut, LG, mtART, mtMAM, mtREV, mtZOA, mtMet, mtVer,<br>
mtInv, Poisson, PMB, rtREV, VT, WAG</p></div></div></div><div class="heading-wrapper"><h5 data-heading="+FreqType碱基使用偏好：Base frequencies 可选设置：" dir="auto" class="heading" id="+FreqType碱基使用偏好：Base_frequencies_可选设置："><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>+FreqType碱基使用偏好：Base frequencies 可选设置：</h5><div class="heading-children"><div><p dir="auto">每个核苷酸位点上的替代是随机发生的，则A,T,C,G出现的频率应该大致相等。实际情况：DNA受到自然选择的压力，各个位点的碱基出现频率并不相等。</p></div><div><p dir="auto"><img alt="图片" src="https://mmbiz.qpic.cn/mmbiz_png/bjvWWMcEDln3nqZicVREABdiayxb9qcQpCfhy36EErP0R4ykdUyax3MpVqlst5zSVkMibsXKj9OA1gw9rCm6JSib4w/640?wx_fmt=png&amp;tp=wxpic&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" referrerpolicy="no-referrer"></p></div><div><p dir="auto">+RateType：rate heterogeneity across sites可选设置：</p></div><div><p dir="auto"><img alt="图片" src="https://mmbiz.qpic.cn/mmbiz_png/bjvWWMcEDln3nqZicVREABdiayxb9qcQpC4CAFcUQMACLF5uBjQLtPm4Pk90nq0QeW5x8mFevne0Ev4zS6dzZ4aQ/640?wx_fmt=png&amp;tp=wxpic&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" referrerpolicy="no-referrer"></p></div><div><p dir="auto">指定替换模型构建进化树命令举例：</p></div><div><pre><code>iqtree&nbsp;-s&nbsp;example.phy&nbsp;-m&nbsp;TIM2+I+G
</code><button class="copy-code-button">复制</button></pre></div></div></div><div class="heading-wrapper"><h5 data-heading="Bootstrap法评估分支支持度" dir="auto" class="heading" id="Bootstrap法评估分支支持度"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>Bootstrap法评估分支支持度</h5><div class="heading-children"><div><p dir="auto">真实的进化信息只有一个，而我们总是拿着有限的序列信息，希望去获得他。能否获得他，是一个问题。而我们使用的序列信息是否能真实且稳定地反应一个进化信息，那么是另外一个事情。bootstrap法常用的，尤其是ML法构建进化树上，分支可靠性检验方法。但是这个计算逻辑最大的问题在于，抽样重新跑，抽样再重新跑，不断重复，直到收敛或者是到指定的比如1000次。计算量大，耗时长。IQ-tree的作者团队在前述提出了一个快速的BS方法，最后整合到IQ-tree中。使用的方式是</p></div><div><pre><code>iqtree&nbsp;-s&nbsp;example.phy&nbsp;-m&nbsp;TIM2+I+G&nbsp;-b&nbsp;1000
</code><button class="copy-code-button">复制</button></pre></div></div></div><div class="heading-wrapper"><h5 data-heading="超快（ultrafast bootstraping）" dir="auto" class="heading" id="超快（ultrafast_bootstraping）"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>超快（ultrafast bootstraping）</h5><div class="heading-children"><div><p dir="auto">大概是IQTREE的精髓所在。顾名思义，ultrafast bootstrap approximation的特点就是超快。这里涉及到的细节，感兴趣的读者可以参见IQTREE的开发者写的几篇文章。作者认为，UFBoot is 10 to 40 times faster than RAxML rapid bootstrap and obtains less biased support values。</p></div><div><pre><code>iqtree&nbsp;-s&nbsp;example.phy&nbsp;-m&nbsp;TIM2+I+G&nbsp;-B&nbsp;1000
</code><button class="copy-code-button">复制</button></pre></div><div><p dir="auto">除ultrafast bootstrap之外，IQTREE还提供了以下检验树的拓扑结构可信度的方法。</p></div><div><ul>
<li data-line="0" dir="auto">-alrt：SH-aLRT检验(4)，没记错的话FastTree2使用的就是这个吧？<br>
</li>
<li data-line="2" dir="auto">-abayes：approximate Bayes test，由瑞士苏黎世应用科学教授Maria Anisimova提出(5)<br>
</li>
<li data-line="4" dir="auto">-lbp：fast local bootstrap probability method，由Adachi and Hasegawa提出（<a data-tooltip-position="top" aria-label="http://www.is.titech.ac.jp/~shimo/class/doc/csm96.pdf%EF%BC%89" rel="noopener" class="external-link" href="http://www.is.titech.ac.jp/~shimo/class/doc/csm96.pdf%EF%BC%89" target="_blank">http://www.is.titech.ac.jp/~shimo/class/doc/csm96.pdf）</a><br>
</li>
</ul></div><div><pre><code>iqtree&nbsp;-s&nbsp;example.phy&nbsp;-m&nbsp;TIM2+I+G&nbsp;-B&nbsp;1000&nbsp;-alrt&nbsp;1000
</code><button class="copy-code-button">复制</button></pre></div><div><p dir="auto">如果你指定了多个检验方法，那么其结果会在树里（.treefile）呈现出来，不同检验所得数值用斜线隔开，比如：((a,b)100/98:0.1,c:0.2)90/95</p></div><div><p dir="auto">最后，iqtree做群体遗传进化构建进化树推荐命令：</p></div><div><pre><code>iqtree2&nbsp;-s&nbsp;supergene.phy&nbsp;-st&nbsp;DNA&nbsp;-T&nbsp;2&nbsp;&nbsp;-mem&nbsp;8G&nbsp;\&nbsp;&nbsp;&nbsp;&nbsp;-m&nbsp;&nbsp;GTR&nbsp;&nbsp;-redo&nbsp;\&nbsp;&nbsp;&nbsp;&nbsp;-B&nbsp;1000&nbsp;-bnni&nbsp;\&nbsp;&nbsp;&nbsp;&nbsp;--prefix&nbsp;iqtree&nbsp;
</code><button class="copy-code-button">复制</button></pre></div></div></div></div></div><div class="heading-wrapper"><h3 data-heading="3.&nbsp;RAxML 构建进化树" dir="auto" class="heading" id="3.&nbsp;RAxML_构建进化树"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>3.&nbsp;RAxML 构建进化树</h3><div class="heading-children"><div><p dir="auto">RAxML是最大似然法（maximum likelihood）建树的经典工具，其由来自德国海德堡理论科学研究所（Heidelberg Institute for Theoretical Studies）的Alexandros Stamatakis开发,最新已经更新了版本RAxML-NG，支持的替换模型更多，运行速度更快。</p></div><div class="heading-wrapper"><h5 data-heading="RAxML 建树原理" dir="auto" class="heading" id="RAxML_建树原理"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>RAxML 建树原理</h5><div class="heading-children"><div><p dir="auto">RAxML采用最大似然法建树，即将系统树的拓扑结构、分支长度及进化模型等的全部或者部分作为需要估计的参数，在给定的数据集及进化模型的基础上，用最大似然法的标准-似然值最大化来估计这些参数。首先，要选择进化模型，以简约树或者联接树为基础，采用似然法估计模型中各个参数。设置好参数后，以简约树或者联接树作为起始树，进行似然分析，最后用统计学方法从多个似然树中寻找最佳得分树。</p></div></div></div><div class="heading-wrapper"><h5 data-heading="RAxML-NG&nbsp;使用" dir="auto" class="heading" id="RAxML-NG&nbsp;使用"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>RAxML-NG&nbsp;使用</h5><div class="heading-children"><div><p dir="auto">RAxML软件支持输入文件的格式可以是比对好的fasta格式或者phylip格式，例如DNA比对序列，核苷酸替代模型为GTR，rate heterogeneity设置为gamma分布，不做bootstraping，命令如下：</p></div><div><pre><code>raxml-ng&nbsp;&nbsp;--msa&nbsp;supergene.phy&nbsp;--model&nbsp;GTR+G&nbsp;&nbsp;--prefix&nbsp;raxml_tree&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;--threads&nbsp;2&nbsp;--seed&nbsp;123
</code><button class="copy-code-button">复制</button></pre></div><div><p dir="auto">如果是建树和bootstrap一起做，可以加--all参数一步完成：</p></div><div><pre><code>raxml-ng&nbsp;&nbsp;--msa&nbsp;supergene.phy&nbsp;--model&nbsp;GTR+G&nbsp;&nbsp;--prefix&nbsp;raxml_tree&nbsp;--threads&nbsp;2&nbsp;--seed&nbsp;123&nbsp;&nbsp;&nbsp;
</code><button class="copy-code-button">复制</button></pre></div><div><p dir="auto">关于raxml-ng模型的设置与iq-tree类似，支持很多种个性化的模型设置：</p></div><div><p dir="auto"><img alt="图片" src="https://mmbiz.qpic.cn/mmbiz_png/bjvWWMcEDllRjKyOZo0rc7kWeAKLnYaKXNLALJgqBa5JibO42u60UichzzOnhOpHV6PE7efFl73ovvp0icibDKpGYg/640?wx_fmt=png&amp;wxfrom=13&amp;tp=wxpic" referrerpolicy="no-referrer"></p></div><div><p dir="auto">详细设置方法见：<a rel="noopener" class="external-link" href="https://github.com/amkozlov/raxml-ng/wiki/Input-data#evolutionary-model" target="_blank">https://github.com/amkozlov/raxml-ng/wiki/Input-data#evolutionary-model</a></p></div><div class="mod-footer"></div></div></div></div></div></div></div></div><div class="sidebar-right sidebar"><div class="sidebar-handle"></div><div class="sidebar-topbar"><div class="topbar-content"></div><div class="clickable-icon sidebar-collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="100%" height="100%" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="3" stroke-linecap="round" stroke-linejoin="round" class="svg-icon"><path d="M21 3H3C1.89543 3 1 3.89543 1 5V19C1 20.1046 1.89543 21 3 21H21C22.1046 21 23 20.1046 23 19V5C23 3.89543 22.1046 3 21 3Z"></path><path d="M10 4V20"></path><path d="M4 7H7"></path><path d="M4 10H7"></path><path d="M4 13H7"></path></svg></div></div><div class="sidebar-content"><div class="graph-view-wrapper"><div class="sidebar-section-header">Interactive Graph</div><div class="graph-view-placeholder">
		<div class="graph-view-container">
			<div class="graph-icon graph-expand" role="button" aria-label="Expand" data-tooltip-position="top"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon"><line x1="7" y1="17" x2="17" y2="7"></line><polyline points="7 7 17 7 17 17"></polyline></svg></div>
			<canvas id="graph-canvas" class="hide" width="512px" height="512px"></canvas>
		</div>
		</div></div><div class="tree-container mod-root nav-folder tree-item outline-tree" data-depth="0"><div class="tree-header"><span class="sidebar-section-header">Table Of Contents</span><button class="clickable-icon collapse-tree-button" aria-label="Collapse All"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"></svg></button></div><div class="tree-scroll-area tree-item-children nav-folder-children"><div class="tree-item mod-tree-folder nav-folder mod-collapsible is-collapsed" style="display: none;"></div><div class="tree-item" data-depth="1"><a class="tree-link" href="软件操作\分析软件\似然树构建.html#似然树构建"><div class="tree-item-contents heading-link" heading-name="似然树构建"><span class="tree-item-title">似然树构建</span></div></a><div class="tree-item-children nav-folder-children"><div class="tree-item" data-depth="3"><a class="tree-link" href="软件操作\分析软件\似然树构建.html#构建进化树的软件"><div class="tree-item-contents heading-link" heading-name="构建进化树的软件"><span class="tree-item-title">构建进化树的软件</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="3"><a class="tree-link" href="软件操作\分析软件\似然树构建.html#1._FastTree_构建进化树"><div class="tree-item-contents heading-link" heading-name="1. FastTree 构建进化树"><span class="tree-item-title">1. 
FastTree 构建进化树
</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="软件操作\分析软件\似然树构建.html#替换模型选择"><div class="tree-item-contents heading-link" heading-name="替换模型选择"><span class="tree-item-title">替换模型选择</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="3"><a class="tree-link" href="软件操作\分析软件\似然树构建.html#2.&nbsp;IQ-tree_构建进化树"><div class="tree-item-contents heading-link" heading-name="2.&nbsp;IQ-tree 构建进化树"><span class="tree-item-title">2.&nbsp;IQ-tree 构建进化树</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="软件操作\分析软件\似然树构建.html#模型选择"><div class="tree-item-contents heading-link" heading-name="模型选择"><span class="tree-item-title">模型选择</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="6"><a class="tree-link" href="软件操作\分析软件\似然树构建.html#指定模型参数设置格式：-m_MODEL+FreqType+RateType，"><div class="tree-item-contents heading-link" heading-name="指定模型参数设置格式：-m MODEL+FreqType+RateType，"><span class="tree-item-title">指定模型参数设置格式：-m MODEL+FreqType+RateType，</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="软件操作\分析软件\似然树构建.html#替换模型MODEL包括："><div class="tree-item-contents heading-link" heading-name="替换模型MODEL包括："><span class="tree-item-title">替换模型MODEL包括：</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="软件操作\分析软件\似然树构建.html#+FreqType碱基使用偏好：Base_frequencies_可选设置："><div class="tree-item-contents heading-link" heading-name="+FreqType碱基使用偏好：Base frequencies 可选设置："><span class="tree-item-title">+FreqType碱基使用偏好：Base frequencies 可选设置：</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="软件操作\分析软件\似然树构建.html#Bootstrap法评估分支支持度"><div class="tree-item-contents heading-link" heading-name="Bootstrap法评估分支支持度"><span class="tree-item-title">Bootstrap法评估分支支持度</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="软件操作\分析软件\似然树构建.html#超快（ultrafast_bootstraping）"><div class="tree-item-contents heading-link" heading-name="超快（ultrafast bootstraping）"><span class="tree-item-title">超快（ultrafast bootstraping）</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="3"><a class="tree-link" href="软件操作\分析软件\似然树构建.html#3.&nbsp;RAxML_构建进化树"><div class="tree-item-contents heading-link" heading-name="3.&nbsp;RAxML 构建进化树"><span class="tree-item-title">3.&nbsp;RAxML 构建进化树</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="软件操作\分析软件\似然树构建.html#RAxML_建树原理"><div class="tree-item-contents heading-link" heading-name="RAxML 建树原理"><span class="tree-item-title">RAxML 建树原理</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="软件操作\分析软件\似然树构建.html#RAxML-NG&nbsp;使用"><div class="tree-item-contents heading-link" heading-name="RAxML-NG&nbsp;使用"><span class="tree-item-title">RAxML-NG&nbsp;使用</span></div></a><div class="tree-item-children nav-folder-children"></div></div></div></div></div></div></div><script defer="">let rs = document.querySelector(".sidebar-right"); rs.classList.add("is-collapsed"); if (window.innerWidth > 768) rs.classList.remove("is-collapsed"); rs.style.setProperty("--sidebar-width", localStorage.getItem("sidebar-right-width"));</script></div></div></body></html>